css/css-backgrounds/border-image-slice-005.htm

```diff
@@ -1,25 +1,29 @@
-<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
-<html xmlns="http://www.w3.org/1999/xhtml">
-    <head>
+<!DOCTYPE html>
+
+  <meta charset="UTF-8">
+
         <title>CSS Test: The 'border-image-slice' property with four percentage values</title>
-        <link rel="author" title="Microsoft" href="http://www.microsoft.com/" />
-        <link rel="help" href="http://www.w3.org/TR/css3-background/#the-border-image-slice" />
-        <meta name="flags" content="image" />
-        <meta name="assert" content="This test checks that the border image is sliced into nine regions with inward offsets, '40%' from the top, '15%' from the right,'20%' from the bottom, and '5%' from the left edges of the image. Percentages are relative to the size of the image: the width of the image for the horizontal offsets, the height for vertical offsets." />
-        <style type="text/css">
+        <link rel="author" title="Microsoft" href="http://www.microsoft.com/">
+        <!--
+        Test rehabilitated by Gérard Talbot
+        2020-05-27
+        -->
+        <link rel="help" href="http://www.w3.org/TR/css3-background/#the-border-image-slice">
+        <link rel="match" href="reference/border-image-slice-005-ref.html">
+        <meta name="flags" content="">
+        <meta name="assert" content="This test checks that the border image is sliced into nine regions with inward offsets, '40%' from the top, '15%' from the right,'20%' from the bottom, and '5%' from the left edges of the image. Percentages are relative to the size of the image: the width of the image for the horizontal offsets, the height for vertical offsets.">
+        <style>
             div
             {
                 border: 40px double red;
                 border-image-slice: 40% 15% 20% 5%;
-                border-image-source: url("../support/9grid40-30-20-10-red.png");
+                border-image-source: url("support/9grid40-30-20-10-red.png");
```

Comment:
So, I know this is not you fault but the fault of the original author... but this image is too simple to actually check for the proper condition. If the UA only slices off the first 1px off each side, it will still render correctly. We need an image that doesn't only detect when the UA slices too much, but also if it slices off too little.

My suggestion here is to use a skewed "striped border": design the image so that once it is sliced and squished into the border, it creates an evenly striped border. Since the numbers here are all divisible by a factor of 5, then 5 even stripes on each side would work well. The image will be uneven from side to side, but once it is squished into the border it will be even all the way around.

If there are stripes missing in the test, then we know the UA sliced too little of the image. If there is red in the test, then it sliced too much. It has to slice exactly the right amount.

Reply:
fantasai, I need more time to carefully read your comment. Right now, I do not know what "skewed 'striped border'", "evenly striped border" and what "squished" mean.

"skewed striped border" == "bordure rayée asymétrique"

"evenly striped border" == "bordure uniformément rayée"

"squished" == "écrasé"

but I do understand that the test would be ideal (at least, more challenging) if an image was designed in a way that display failures if the UA slices too much or slices too little.